src/components/Audio.cpp

```diff
@@ -178,51 +178,115 @@ bool Audio::setRate(double rate)
 
 void Audio::mix(const int16_t* samples, size_t frames)
 {
-  _logger->debug(TAG "Requested %zu audio frames", frames);
+  _logger->debug(TAG "Processing %zu audio frames", frames);
 
   size_t avail = _fifo->free();
+  size_t output_size;
+  int16_t* output;
+  spx_uint32_t out_len;
 
-  /* Readjust the audio input rate. */
```

Comment:
This is also unclear to me now, but I seem to remember that the emulation would be run locked to vsync to avoid tearing and/or hiccups. When the core is run in a different rate than it's supposed to run, the audio must be resampled because it'll either be too much or too few. In actuality, given slight variations both in the vsync and audio sample rate hardware times, I was under the impression that resampling would always be necessary to account for these variations.